packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -0,0 +1,134 @@
+
+  @override
+  State<ProductImageViewer> createState() => _ProductImageViewerState();
+}
+
+class _ProductImageViewerState extends State<ProductImageViewer> {
+  late final ProductImageData imageData;
+
+  /// When the image is edited, this is the new image
+  ImageProvider? imageProvider;
+  bool _isEdited = false;
+
+  @override
+  void initState() {
+    imageData = widget.imageData;
+
+    if (imageData.imageUrl != null) {
+      imageProvider = NetworkImage(imageData.imageUrl!);
+    }
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) => SmoothScaffold(
+        extendBodyBehindAppBar: true,
+        backgroundColor: Colors.black,
+        floatingActionButton: createEditButton(onPressed: _editImage),
+        appBar: AppBar(
+          backgroundColor: Colors.black,
+          foregroundColor: WHITE_COLOR,
+          elevation: 0,
+          title: Text(imageData.title),
+          leading: SmoothBackButton(
+            onPressed: () => Navigator.maybePop(context, _isEdited),
+          ),
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            ConstrainedBox(
+              constraints: BoxConstraints.tight(
+                Size(double.infinity, MediaQuery.of(context).size.height / 2),
+              ),
+              child: PhotoView(
+                minScale: 0.2,
+                imageProvider: imageProvider,
+                heroAttributes: PhotoViewHeroAttributes(
+                  tag: imageProvider!,
+                ),
+                backgroundDecoration: const BoxDecoration(
+                  color: Colors.black,
+                ),
+              ),
+            ),
+          ],
+        ),
+      );
+
+  Future<File> _downloadImageFile(String url) async {
+    final http.Response response = await http.get(Uri.parse(url));
+    final Directory tempDirectory = await getTemporaryDirectory();
+    final File imageFile = await File('${tempDirectory.path}/editing_image')
+        .writeAsBytes(response.bodyBytes);
+    return imageFile;
+  }
+
+  Future<void> _editImage() async {
+    final File? imageFile = await LoadingDialog.run<File>(
+      context: context,
+      future: _downloadImageFile(imageData.imageUrl!),
+    );
+
+    if (imageFile == null) {
+      return;
+    }
+
+    if (!mounted) {
+      return;
+    }
+
+    final File? photoUploaded = await Navigator.push<File?>(
+      context,
+      MaterialPageRoute<File?>(
+        builder: (BuildContext context) => ConfirmAndUploadPicture(
+          barcode: widget.barcode,
+          imageType: imageData.imageField,
+          initialPhoto: imageFile,
+        ),
+      ),
+    );
+    if (photoUploaded != null) {
+      _isEdited = true;
+      if (!mounted) {
+        return;
+      }
+
+      setState(() {
+        imageProvider = FileImage(photoUploaded);
+      });
+    }
+  }
+
+  Widget createEditButton({required Function() onPressed}) =>
```

Comment:
Should be private. Or, as it used only once and is not very complex, the code should be put where it is called (I mean, without a method).

Reply:
I'm sorry, I thought I made it private but that's not the case 😆